packages/ra-ui-materialui/src/input/AutocompleteInput.stories.tsx

```diff
@@ -326,61 +336,93 @@ export const OnCreateSlow = () => (
     </Wrapper>
 );
 
-export const OnCreatePrompt = () => (
-    <Wrapper>
+const OnCreatePromptInput = () => {
+    const [choices, setChoices] = useState(choicesForCreationSupport);
+    return (
         <AutocompleteInput
             source="author"
-            choices={choicesForCreationSupport}
-            onCreate={filter => {
+            choices={choices}
+            onCreate={async filter => {
                 const newAuthorName = window.prompt(
                     'Enter a new author',
                     filter
                 );
-
-                if (newAuthorName) {
-                    const newAuthor = {
-                        id: choicesForCreationSupport.length + 1,
-                        name: newAuthorName,
-                    };
-                    choicesForCreationSupport.push(newAuthor);
-                    return newAuthor;
-                }
+                if (!newAuthorName) return;
+                const newAuthor = {
+                    id: choices.length + 1,
+                    name: newAuthorName,
+                };
+                setChoices(authors => [...authors, newAuthor]);
```

Comment:
We still have a brief flash issue on option create in this story

Reply:
Fixed